src/DomainValidator.php

```diff
@@ -271,8 +271,13 @@ protected function validateValue($value)
             }
         }
 
-        if ($this->checkDNS && !checkdnsrr("$value.", 'ANY')) {
-            return $this->getErrorMessage('messageDNS');
+        if ($this->checkDNS) {
+            $dnsRecords = dns_get_record("$value.", DNS_ANY);
```

Comment:
I think we should not call dns_get_record here. If developer wants to use callable then he probably wants to do his own check and not interested in results of dns_get_record("$value.", DNS_ANY). Developer which needs dns_get_record is still able to call it in callable. Also I would like to make signature of callable is simpler. What about such approach:
```php
if (is_callable($this->checkDNS)) {
    return call_user_func($value);
}

if (empty(dns_get_record("$value.", DNS_ANY))) {
    return $this->getErrorMessage('messageDNS');
}
```

Also please change access rights for changed files back to 644.

Reply:
Fixed errors in my previous comment.